### PR TITLE
Enable project and workspace creation

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
- leverage existing code to create project and workspaces on init
- add Stage workspace on project creation
- enable runtime on every newly created worskpace
- extend autocomplete plugin to support an escape key for creation ('+')

Relates to: https://github.com/adobe/aio-cli-plugin-app/pull/319
Depends on: https://github.com/adobe/aio-lib-console/pull/15

NEEDS A RELEASE IN AIO-LIB-CONSOLE + Major dep bump before merging this

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -128,35 +134,48 @@ class ConsoleGenerator extends Generator {
...
     spinner.start()
 
-    spinner.text = 'Getting projects...'
+    spinner.text = 'Getting Projects...'
     const projects = (await this.sdkClient.getProjectsForOrg(orgId)).body
     spinner.stop()
 
-    const projectsList = projects.map(item => item.title)
+    // show projects by title and reverse order to show latest first
+    const projectsList = projects.map(item => item.title).reverse()
```
Comment (shazron) [inline generators/app/index.js:142]: note that .reverse() modifies the projects object here as well (in place reverse), FYI. Not sure if it affects things later

File: generators/app/index.js [hunk 2]
```diff
@@ -128,35 +134,48 @@ class ConsoleGenerator extends Generator {
...
+      const projectId = createdProject.projectId
+
+      // create the missing stage workspace
+      spinner.text = 'Creating Stage Workspace...'
+      await this.sdkClient.createWorkspace(orgId, projectId, { name: 'Stage' })
+
+      // enable runtime on the Production and Stage workspace
+      spinner.text = 'Enabling Adobe I/O Runtime...'
+      const workspaces = (await (this.sdkClient.getWorkspacesForProject(orgId, projectId))).body
+      await Promise.all(workspaces.map(w => this.sdkClient.createRuntimeNamespace(orgId, projectId, w.id)))
```
Comment (shazron) [inline generators/app/index.js:174]: This is a comment for everything in the project and workspace creation. What happens when there is an exception somewhere, there could be a project created half-baked, and we don't delete it, correct? 

We don't have a concept of "transactions" for creating a project, perhaps something to keep in mind for the future, or suggest that all this be handled in the Console API for the future.

File: lib/validate.js [hunk 3]
```diff
@@ -9,19 +9,22 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-  validateName,
-  validateTitle
+  // project input validators
+  validateProjectName: createValidator('Project name', /^[a-zA-Z0-9]{1,20}$/),
+  validateProjectTitle: createValidator('Project title', /^[\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{1,45}$/),
+  validateProjectDescription: createValidator('Project description', /^.{0,1000}$/), // can be empty
+  // workspace input validators
+  validateWorkspaceName: createValidator('Workspace name', /^[a-zA-Z0-9]{1,20}$/),
+  validateWorkspaceTitle: createValidator('Workspace title', /^[\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{0,45}$/) // can be empty
+  // validateWorkspaceDescription: createValidator('Workspace description', /^(?=\S).{0,500}(?<=\S)$/)
```
Comment (shazron) [inline lib/validate.js:29]: i assume there is no restrictions for description?

Comment (moritzraho) [reply lib/validate.js:29]: for projects is 1000 chars max, and 500 for workspaces however we don't asked for a description in workspaces, should we ?

File: generators/app/index.js (hunk 2 above)
Comment (moritzraho) [reply generators/app/index.js:174]: very good point, yes this is correct
will try to fix that, would be good to suggest it to the Console API too

File: generators/app/index.js (hunk 1 above)
Comment (moritzraho) [reply generators/app/index.js:142]: ok thanks for the heads up it shouldn't change much but will double check

File: lib/validate.js (hunk 3 above)
Comment (shazron) [reply lib/validate.js:29]: true, so no need for validation there

File: generators/app/index.js (hunk 2 above)
Comment (shazron) [reply generators/app/index.js:174]: we can push this off as a new feature

Comment (moritzraho) [reply generators/app/index.js:174]: ok, + I started working on this and realized deleteRuntimeNamespace doesn't exist yet so we can't really cleanup things fully. Will open an issue after we merge this

File: generators/app/index.js (hunk 1 above)
Comment (moritzraho) [reply generators/app/index.js:142]: ok applied reverse on a copy

**Comment by moritzraho** [comment]
@shazron comments resolved 🙂

**Review by shazron — APPROVED** [review]

**Pull request merged**